main.tf

```diff
@@ -6,7 +6,7 @@ data "aws_caller_identity" "current" {}
 ################################################################################
 
 resource "aws_kms_key" "this" {
-  count = var.create && !var.create_external ? 1 : 0
+  count = var.create && !var.create_external && !var.create_replica && !var.create_replica_external ? 1 : 0
```

Comment:
to simplify this and based on the comments below, maybe we just add another variable `var.create_key` which defaults to true